File: backend/src/main/java/mn_react/core/domain/entities/Loan.java
package mn_react.core.domain.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Loan {
    private Long id;
    private Long userId;
    private Long bookId;
    private LocalDate dataEmprestimo;
    private LocalDate dataPrevistaDevolucao;
    private LocalDate dataRealDevolucao;
    private LoanStatus status;

    public boolean isAtivo() {
        return status == LoanStatus.ATIVO;
    }

    public boolean isDevolvido() {
        return status == LoanStatus.DEVOLVIDO;
    }

    public boolean isAtrasado() {
        return status == LoanStatus.ATRASADO;
    }
}

File: frontend/src/features/loans/shared/types/Loan.ts
export type TLoanStatus = 'ATIVO' | 'DEVOLVIDO' | 'ATRASADO';

export interface ILoan {
  id: number;
  userId: number;
  bookId: number;
  dataEmprestimo: string;
  dataPrevistaDevolucao: string;
  dataRealDevolucao?: string;
  status: TLoanStatus;
}

export interface ILoanFilters {
  page: number;
  size: number;
  status?: TLoanStatus;
  userId?: number;
}

export interface ICreateLoanRequest {
  userId: number;
  bookId: number;
  dataPrevistaDevolucao?: string;
}
